Makefile

```
@@ -32,6 +31,6 @@ clean::
 	rm -f *.o libtree
 
 clean check::
-	for dir in $(sort $(wildcard tests/*)); do \
-		$(MAKE) -C $$dir $@ || break ;\
+	ls -1 -d tests/* | while read dir; do \
```

```suggestion
	find tests -type d -mindepth 1 -maxdepth 1 | while read -r dir; do \
```

squashes https://github.com/koalaman/shellcheck/wiki/SC2012 and https://github.com/koalaman/shellcheck/wiki/SC2162